src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java

```diff
@@ -634,13 +636,117 @@ CosmosDocumentList find(String coll, Aggregate aggregate, Condition cond, String
 			log.info("find Document:{}, cond:{}, partition:{}, account:{}", collectionLink, cond, cond.crossPartition ? "crossPartition" : partition, getAccount());
 		}
 
-		var jsonObjs = docs.stream().map(it -> it.toObject(JSONObject.class)).collect(Collectors.toList());
+        if(Objects.isNull(aggregate)&&!cond.joinCondText.isEmpty()&&!cond.returnAll){
+            jsonObjs=mergeListToDoc(cond,docs,collectionLink, querySpec, feedOptions);
+        }
+        else{
+            jsonObjs=docs.stream().map(it -> it.toObject(JSONObject.class)).collect(Collectors.toList());
+        }
 
 		return new CosmosDocumentList(jsonObjs);
 
 	}
 
-	/**
+    /**
+     * Merge the list to origin list
+     * This function will traverse the result of join part and replaced by new result that is found by sub query.
+     * @param cond merge the content of the list to origin list
+     * @param docs origin docs
+     * @param collectionLink collection link
+     * @param querySpec querySpec
+     * @param feedOptions feed Options
+     * @return docs list
+     * @throws Exception error exception
+     */
+    private List<JSONObject> mergeListToDoc(Condition cond, List<Document> docs, String collectionLink, SqlQuerySpec querySpec, FeedOptions feedOptions) throws Exception {
+
+        Map<String,String[]> keyMap=new LinkedHashMap<>();
+
+        var queryText=initJoinSelectPart(cond,querySpec,keyMap);
+        var subDocs = RetryUtil.executeWithRetry(() -> client.queryDocuments(collectionLink, new SqlQuerySpec(queryText, querySpec.getParameters()), feedOptions).getQueryIterable().toList());
```

Comment:
It is werid that we need a db query one more time here.
We just want to filter out some results out, which have already been fetched in th e previous query.

Expect: no more db queries.